x-pack/plugins/uptime/public/components/settings/alert_defaults_form.tsx

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useEffect, useState, useRef, useCallback } from 'react';
+import { FormattedMessage } from '@kbn/i18n/react';
+import {
+  EuiDescribedFormGroup,
+  EuiFormRow,
+  EuiTitle,
+  EuiSpacer,
+  EuiComboBox,
+  EuiText,
+  EuiIcon,
+  EuiComboBoxOptionOption,
+} from '@elastic/eui';
+import { useSelector } from 'react-redux';
+import { SettingsFormProps } from '../../pages/settings';
+import { connectorsSelector } from '../../state/alerts/alerts';
+import { AddConnectorFlyout } from './add_connector_flyout';
+import { useGetUrlParams, useUrlParams } from '../../hooks';
+import { getActionTypeIcon } from '../../../../triggers_actions_ui/public';
+import { alertFormI18n } from './translations';
+import { useInitApp } from '../../hooks/use_init_app';
+
+type ConnectorOption = EuiComboBoxOptionOption<string>;
+
+export const AlertDefaultsForm: React.FC<SettingsFormProps> = ({
+  onChange,
+  loading,
+  formFields,
+  fieldErrors,
+  isDisabled,
+}) => {
+  const { focusConnectorField } = useGetUrlParams();
+
+  const updateUrlParams = useUrlParams()[1];
+
+  const inputRef = useRef<HTMLInputElement | null>(null);
+
+  useInitApp();
+
+  useEffect(() => {
+    if (focusConnectorField && inputRef.current && !loading) {
+      inputRef.current.focus();
+    }
+  }, [focusConnectorField, inputRef, loading]);
+
+  const { data = [] } = useSelector(connectorsSelector);
+
+  const [error, setError] = useState<string | undefined>(undefined);
+
+  const onBlur = () => {
+    if (inputRef.current) {
+      const { value } = inputRef.current;
+      setError(value.length === 0 ? undefined : `"${value}" is not a valid option`);
+    }
+    if (inputRef.current && !loading && focusConnectorField) {
+      updateUrlParams({ focusConnectorField: undefined });
+    }
+  };
+
+  const onSearchChange = (value: string, hasMatchingOptions?: boolean) => {
+    setError(
+      value.length === 0 || hasMatchingOptions ? undefined : `"${value}" is not a valid option`
+    );
+  };
+
+  const options = (data ?? []).map((connectorAction) => ({
+    value: connectorAction.id,
+    label: connectorAction.name,
+    'data-test-subj': connectorAction.name,
+  }));
+
+  const renderOption = (option: ConnectorOption) => {
+    const { label, value } = option;
+
+    const { actionTypeId: type } = data?.find((dt) => dt.id === value) ?? {};
+    return (
+      <EuiText size="s">
+        <EuiIcon type={getActionTypeIcon(type as string)} /> <span>{label}</span>
```

Comment:
I took a look to see if this can be done in a more dynamic way (once plugins start adding their own 3rd party action types) and I think using the `actionTypeRegistry` exposed by the `triggers_actions_ui` plugin's start contract may do the trick? Let me know!

`startPlugins.triggers_actions_ui.actionTypeRegistry.get('.slack').iconClass`

Reply:
done, this looks good